File: packages/csharp/src/Errors/BankUnderMaintenanceException.cs
using System;

namespace Tesote.Sdk.Errors;

/// <summary>503 BANK_UNDER_MAINTENANCE — upstream bank is unavailable.</summary>
public sealed class BankUnderMaintenanceException : ApiException
{
    /// <summary>Construct with the full required-field set.</summary>
    public BankUnderMaintenanceException(
        string? message, string? errorCode, int httpStatus,
        string? requestId, string? errorId, int? retryAfter,
        string? responseBody, RequestSummary? requestSummary,
        int attempts, Exception? cause)
        : base(message, errorCode, httpStatus, requestId, errorId,
               retryAfter, responseBody, requestSummary, attempts, cause)
    {
    }
}

File: packages/java/src/main/java/com/tesote/sdk/errors/BankUnderMaintenanceException.java
package com.tesote.sdk.errors;

/**
 * 503 — the upstream bank reported maintenance. Retry after the configured
 * window. Distinct from the broader {@link ServiceUnavailableException}
 * (which signals API-side pause mode).
 */
public final class BankUnderMaintenanceException extends ApiException {
    public BankUnderMaintenanceException(String message, String errorCode, int httpStatus,
                                         String requestId, String errorId, Integer retryAfter,
                                         String responseBody, RequestSummary requestSummary,
                                         int attempts, Throwable cause) {
        super(message, errorCode, httpStatus, requestId, errorId,
                retryAfter, responseBody, requestSummary, attempts, cause);
    }
}
